src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -312,14 +312,7 @@ parseLeaf(
                                 LedgerFormats::getInstance().findTypeByName(
                                     strValue));
 
-                            if (!(0u <= type &&
-                                  type <=
-                                      std::min<unsigned>(
-                                          std::numeric_limits<
-                                              std::uint16_t>::max(),
-                                          std::numeric_limits<
-                                              std::underlying_type_t<
-                                                  LedgerEntryType>>::max())))
+                            if (type < 1 || type >= 32768)
```

Comment:
How about `std::numeric_limits<std::underlying_type_t<LedgerEntryType>>::max() - 1`? or with `std::uint16_t` instead of `std::underlying_type_t<LedgerEntryType>`?

Reply:
The problem is that anything over 2<sup>15</sup> is not valid in this context, which is why I changed this code. I hear what you're saying thought; these are just magic numbers, pulled out of thin air, as far as the reader is concerned.

I wondered whether this is an improvement:

```
inline bool isValidLedgerEntryType(std::uint16_t value)
{
    return (value != 0) && (value < 32768);
}

inline bool isValidLedgerEntryType(LedgerEntryType value)
{
    return isValidLedgerEntryType<std::underlying_type_t<LedgerEntryType>>(value);
}
```
This check also be added to the `LedgerEntryTypesMatch` invariant check; it has an explicit check for all "known" values, but this check would be cheap and add an additional layer of safety.

But, after typing this, it occurs to me that this may be overkill: since these are "protocol defined" minima and maxima we could simply add two constants to **`Protocol.h`**:

```
/** The smallest valid type identifier for a ledger entry. */
std::uint16_t constexpr minLedgerEntryType = 1;

/** The largest valid type identifier for a ledger entry. */
std::uint16_t constexpr maxLedgerEntryType = 32767;
```

Thoughts?